src/sql/workbench/contrib/notebook/browser/notebookExplorer/searchResultsViewPane.ts

```diff
@@ -237,7 +241,53 @@ export class NotebookSearchView extends SearchView {
 		}));
 	}
 
-	public startSearch(query: ITextQuery, excludePatternText: string, includePatternText: string, triggeredOnType: boolean, searchWidget: NotebookSearchWidget): Thenable<void> {
+	public async validateAndSearch(query: ITextQuery, searchWidget: SimpleSearchWidget): Promise<void> {
+		await this.validateQry(query).catch(e => errors.onUnexpectedError(e));
+		if (this.parentContainer.views.length > 1) {
+			let filesToIncludeFiltered: string = '';
+			this.parentContainer.views.forEach(async (v) => {
+				let booksViewPane = (<TreeViewPane>this.parentContainer.getView(v.id));
+				if (booksViewPane?.treeView?.root) {
+					let root = booksViewPane.treeView.root;
+					if (root.children) {
+						let items = root.children;
+						items?.forEach(async root => {
+							searchWidget.updateViewletsState();
+							let folderToSearch: IFolderQuery = { folder: URI.file(path.join(isString(root.tooltip) ? root.tooltip : root.tooltip.value, 'content')) };
+							query.folderQueries.push(folderToSearch);
+							filesToIncludeFiltered = filesToIncludeFiltered + path.join(folderToSearch.folder.fsPath, '**', '*.md') + ',' + path.join(folderToSearch.folder.fsPath, '**', '*.ipynb') + ',';
+							await this.startSearch(query, null, filesToIncludeFiltered, false, searchWidget);
+						});
+					}
+				}
+			});
+		}
+	}
+
+	private async validateQry(query: ITextQuery): Promise<void> {
+		// Validate folderQueries
```

Comment:
I don't really get what this comment is for. And in general please add more comments (and a doc comment) to this function. It's not obvious what "validateQry" would mean unless you go through the code manually.